sdk/core/Azure.Core/src/Shared/ConnectionString.cs

```diff
@@ -0,0 +1,248 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Azure.Core
+{
+    internal sealed class ConnectionString
+    {
+        private readonly Dictionary<string, string> _segments;
+        private readonly string _segmentSeparator;
+        private readonly string _keyValueSeparator;
+
+        public static ConnectionString Parse(string connectionString, string segmentSeparator = ";", string keyValueSeparator = "=", bool allowFlags = false, bool allowEmptyValues = false, bool allowWhitespaces = false)
```

Comment:
Is it necessary to make all of these things configurable?  It adds complexity and length to the code (and this increases the bug surface).  If we don't have evidence that these things would show up in a connection string, I'd make this as minimal as needed for now, given it's an internal class and it would be ok to change the API for it later.